Make plates engine service available

alexander-schranz opened the pull request:
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | no

### Description

This make plates itself available as service and register and create it already correctly configured in the PlatesEngineFactory. This is similar how the mezzio-twigrenderer package works and so make things more consistent and possible to use the plates engine service directly.

<!--
Tell us about why this change is necessary:
- Are you fixing a bug or providing a failing unit test to demonstrate a bug?
  - How do you reproduce it?
  - What did you expect to happen?
  - What actually happened?
  - TARGET THE master BRANCH

- Are you adding documentation?
  - TARGET THE master BRANCH

- Are you providing a QA improvement (additional tests, CS fixes, etc.) that
  does not change behavior?
  - Explain why the changes are necessary
  - TARGET THE master BRANCH

- Are you fixing a BC Break?
  - How do you reproduce it?
  - What was the previous behavior?
  - What is the current behavior?
  - TARGET THE master BRANCH

- Are you adding something the library currently does not support?
  - Why should it be added?
  - What will it enable?
  - How will the code be used?
  - TARGET THE develop BRANCH

- Are you refactoring code?
  - Why do you feel the refactor is necessary?
  - What types of refactoring are you doing?
  - TARGET THE develop BRANCH
-->

alexander-schranz committed:
Make plates engine service available

Signed-off-by: Alexander Schranz <alexander@sulu.io>

Ocramius commented on src/PlatesRendererFactory.php:
```diff
@@ -70,13 +49,21 @@ public function __invoke(ContainerInterface $container): PlatesRenderer
 
-        $engineFactory = new PlatesEngineFactory();
-        return $engineFactory($container);
+        trigger_error(sprintf(
```
IMO, this execution branch can be dropped completely.

We already provide service configuration that works with the declared service.

alexander-schranz replied:
Then I think we can do the same here:

 - https://github.com/mezzio/mezzio-twigrenderer/blob/cfe9ffb3521dab22cafa8e4c3b5865af0f117918/src/TwigRendererFactory.php#L89-L98

alexander-schranz committed:
Drop part when engine is not available

Signed-off-by: Alexander Schranz <alexander@sulu.io>

alexander-schranz committed:
Move some PlatesRendererFactory tests to PlatesEngineFactory tests

Signed-off-by: Alexander Schranz <alexander@sulu.io>

Ocramius commented:
@alexander-schranz please check CI failures 😁

alexander-schranz committed:
Fix code style

Signed-off-by: Alexander Schranz <alexander@sulu.io>

alexander-schranz committed:
Update psalm baseline

Signed-off-by: Alexander Schranz <alexander@sulu.io>

Ocramius commented:
BTW, regenerating the baseline would suffice 👍

alexander-schranz commented:
@Ocramius code style fixed and baseline updated.

alexander-schranz committed:
Use get_class($var) instead of $var::class for php 7.4 support

Signed-off-by: Alexander Schranz <alexander@sulu.io>

alexander-schranz commented:
Not sure about psalm report. When running psalm with the baseline I get no error longer:

<img width="949" alt="Bildschirmfoto 2022-09-20 um 14 51 54" src="https://user-images.githubusercontent.com/1698337/191262611-b1ccd9db-2d3e-4296-8fcb-a7ec3bbe6225.png">

alexander-schranz commented:
Looks for me the baseline is not keep in mind when running without --use-baseline which the ci currently does.

<img width="1071" alt="Bildschirmfoto 2022-09-20 um 15 01 33" src="https://user-images.githubusercontent.com/1698337/191264645-7c90a614-3768-47b6-869c-4d6b7a4c2b90.png">

Not sure how to fix this.

Ocramius commented:
@alexander-schranz I will try to fix this later, and then will report back to explain 👍

Ocramius committed:
Re-generated psalm baseline from a clean PHP 7.4 environment

Used:

```sh
docker run --rm -ti -v $(pwd):/app -w /app php:7.4 vendor/bin/psalm --no-cache --set-baseline=psalm-baseline.xml
```

Ocramius reviewed (approved):
LGTM! Psalm issues are due to code shifting from a class to another, but otherwise nothing new happening :)

Ocramius commented:
Thanks @alexander-schranz! See commit message for resolution.

Also, releasing now.

alexander-schranz commented:
@Ocramius Thank you!
